core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/regionserver/transactional/IdTm.java

```diff
@@ -122,13 +124,67 @@ public void id(int timeout, IdTmId id) throws IdTmException {
               LOG.error("native_id returned: " + err + " Throwing IdTmException");
               throw new IdTmException("ferr=" + err);
            }
+           if (id.val == 0) {
+              LOG.error("native_id returned id: " + id.val + " err: " + err + ", Throwing IdTmException");
```

Comment:
Doesn't hurt anything to report err here,, but you already know in this code path that err == 0 due to the preceding if + throw.